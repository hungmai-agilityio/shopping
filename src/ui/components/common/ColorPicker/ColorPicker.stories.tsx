import { Meta, StoryObj } from '@storybook/react';
import { mockProducts } from '@/mocks';

import { ColorPicker } from '@/ui/components';

const meta: Meta<typeof ColorPicker> = {
  component: ColorPicker,
  parameters: {
    layout: 'centered'
  },
  tags: ['autodocs'],
  argTypes: {
    colors: {
      description: 'List colors'
    },
    isTitle: {
      description: 'Check to show title',
      control: 'boolean'
    },
    titleColor: {
      description: 'Set color for title',
      control: 'text'
    },
    onClick: {
      description: 'Handles color selection'
    }
  }
};

export default meta;

type Story = StoryObj<typeof ColorPicker>;

export const Default: Story = {
  args: {
    colors: mockProducts[0].colors,
    isTitle: true
  }
};
