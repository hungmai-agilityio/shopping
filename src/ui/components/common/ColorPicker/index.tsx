'use client';
import { clsx } from 'clsx';
import { useEffect, useState, memo } from 'react';

// Constants
import { FONT_SIZE, FONT_WEIGHT } from '@/constants';

// Components
import Button from '@/ui/components/common/Button';
import Typography from '@/ui/components/common/Typography';

interface IProps {
  isTitle?: boolean;
  colors: string[];
  onClick?: (color: string) => void;
  titleColor?: string;
}

const ColorPicker = memo(
  ({ isTitle, colors, onClick, titleColor = 'text-black' }: IProps) => {
    const [selectedColor, setSelectedColor] = useState<string>(colors[0]);

    useEffect(() => {
      if (colors.length > 0) {
        setSelectedColor(colors[0]);
        onClick?.(colors[0]);
      }
    }, [colors, onClick]);

    // Handler for color selection
    const handleColorClick = (color: string) => {
      setSelectedColor(color);
      onClick?.(color);
    };

    return (
      <div>
        {isTitle && (
          <Typography
            color={titleColor}
            fontWeight={FONT_WEIGHT.BOLD}
            size={FONT_SIZE.X_SMALL}
          >
            Color
          </Typography>
        )}
        <div className="mt-4 flex items-center gap-3">
          {colors?.map((color) => (
            <Button
              id={color}
              key={color}
              onClick={handleColorClick.bind(null, color)}
              className={clsx(
                `h-4 w-4 rounded-full focus:border focus:border-green-800`,
                {
                  'border border-dark': color === 'White',
                  'ring-2 ring-green-800': color === selectedColor
                }
              )}
              style={{ backgroundColor: color }}
            ></Button>
          ))}
        </div>
      </div>
    );
  }
);

export default ColorPicker;
