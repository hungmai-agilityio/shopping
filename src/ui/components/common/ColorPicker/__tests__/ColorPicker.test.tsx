import { mockProducts } from '@/mocks';
import ColorPicker from '@/ui/components/common/ColorPicker';
import { render, screen } from '@testing-library/react';

describe('Color picker component', () => {
  it('render with default props', () => {
    render(<ColorPicker colors={mockProducts[0].colors} />);
    expect(screen.getByText('Color')).toBeInTheDocument();
  });

  it('should render match to snapshot', () => {
    const { container } = render(
      <ColorPicker colors={mockProducts[0].colors} />
    );
    expect(container).toMatchSnapshot();
  });
});
