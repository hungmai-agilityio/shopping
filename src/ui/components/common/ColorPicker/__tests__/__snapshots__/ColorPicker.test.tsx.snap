// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color picker component should render match to snapshot 1`] = `
<div>
  <div>
    <p
      class="font-bold text-black lg:text-lg text-sm"
    >
      Color
    </p>
    <div
      class="mt-4 flex items-center gap-3"
    >
      <button
        class="h-4 w-4 rounded-full focus:border focus:border-green-800 border border-dark ring-2 ring-green-800"
        id="White"
        style="background-color: White;"
      />
      <button
        class="h-4 w-4 rounded-full focus:border focus:border-green-800"
        id="Blue"
        style="background-color: Blue;"
      />
      <button
        class="h-4 w-4 rounded-full focus:border focus:border-green-800"
        id="Green"
        style="background-color: Green;"
      />
    </div>
  </div>
</div>
`;
